zebrad/src/commands/start.rs

```diff
@@ -253,6 +259,22 @@ impl StartCmd {
 
         exit_status
     }
+
+    /// Returns the bound for the state service buffer,
+    /// based on the configurations of the services that use the state concurrently.
+    fn state_buffer_bound() -> usize {
+        let config = app_config().clone();
+
+        // TODO: do we also need to account for concurrent use across services?
+        //       we could multiply the maximum by 3/2, or add a fixed constant
+        max(
+            config.sync.max_concurrent_block_requests,
+            max(
+                inbound::downloads::MAX_INBOUND_CONCURRENCY,
+                mempool::downloads::MAX_INBOUND_CONCURRENCY,
+            ),
+        )
```

Comment:
Optional: just wanted to point out a suggestion in case it's easier to read:

```suggestion
        [
            config.sync.max_concurrent_block_requests,
            inbound::downloads::MAX_INBOUND_CONCURRENCY,
            mempool::downloads::MAX_INBOUND_CONCURRENCY,
        ]
        .into_iter()
        .max()
        .expect("array is not empty")
```

Reply:
If there was one more entry, it would be a clear winner, but I'm not so sure with just 3.

And I think this code could change soon anyway.